src/core_editor/editor.rs

```diff
@@ -526,7 +535,19 @@ impl Editor {
         self.line_buffer.move_to_end();
     }
 
-    fn cut_selection(&mut self) {
+    fn cut_selection_to_system(&mut self) {
+        #[cfg(not(feature = "system_clipboard"))]
+        panic!("The OS clipboard is not enabled. Enable its use with the reedline/system_clipboard feature!");
+        #[cfg(feature = "system_clipboard")]
+        if let Some((start, end)) = self.get_selection() {
```

Comment:
From the "don't panic" Nushell side this is a dangerous panic footgun.

`EditCommand::Paste { system_clipboard: true }` provides a possibly more aesthetic API over separate variants. But if you were to compile without `system_clipboard` and pass the same event this could panic the shell. 
https://github.com/nushell/reedline/blob/998a2f79092a4c9058e1394a1495063ded8f42ad/src/core_editor/editor.rs#L132-L137
The match compiles either way. This would require extra care while parsing complex user configs.

Either we completely restrict it at compile time via separate variants or turn it into something handlable.

Reply:
I assume turning it into something handlable would involve returning a Result up the call stack so nushell can do something like inform the user? In general I think that might be a good idea, it also opens up for other failable commands should the need arise. Though I don't know if that is the right approach for handling this specific issue. What is your opinion on it? How should it be handled?